src/bin/radclient.c

```diff
@@ -462,10 +461,12 @@ static int radclient_init(TALLOC_CTX *ctx, rc_file_pair_t *files)
 				}
 
 				if (vp->da == attr_packet_type) {
-					vp = fr_cursor_remove(&cursor);	/* so we don't break the filter */
+					fr_pair_t *next;
+					next = fr_pair_list_next(&request->filter, vp);	/* so we don't break the filter */
 					request->filter_code = vp->vp_uint32;
+					fr_pair_remove(&request->filter, vp);
```

Comment:
Removing and freeing are going to be very common patterns.  Could we get an `fr_pair_free` function which takes a list head and the pair to free.  We could actually do the unlinking in the pair destructor and just use `talloc_free` but that has the feeling of a decision we'll regret later.

Reply:
This should have been fr_pair_delete() which does both.